experimental/regression_suite/shark-test-suite-models/sd3/test_clip.py

```diff
@@ -0,0 +1,167 @@
+    sd3_clip_inference_input_1,
+    sd3_clip_inference_input_2,
+    sd3_clip_inference_input_3,
+    sd3_clip_inference_input_4,
+    sd3_clip_inference_input_5,
+    sd3_clip_inference_output_0,
+    sd3_clip_inference_output_1,
+):
+    return [
+        f"--input=1x77x2xi64=@{sd3_clip_inference_input_0.path}",
+        f"--input=1x77x2xi64=@{sd3_clip_inference_input_1.path}",
+        f"--input=1x77x2xi64=@{sd3_clip_inference_input_2.path}",
+        f"--input=1x77x2xi64=@{sd3_clip_inference_input_3.path}",
+        f"--input=1x77x2xi64=@{sd3_clip_inference_input_4.path}",
+        f"--input=1x77x2xi64=@{sd3_clip_inference_input_5.path}",
+        f"--expected_output=2x154x4096xf32=@{sd3_clip_inference_output_0.path}",
+        f"--expected_output=2x2048xf32=@{sd3_clip_inference_output_1.path}",
+    ]
+
+
+ROCM_COMPILE_FLAGS = [
+    "--iree-hal-target-backends=rocm",
+    f"--iree-rocm-target-chip={rocm_chip}",
+    "--iree-input-type=torch",
+    "--iree-opt-const-eval=false",
+    "--iree-global-opt-propagate-transposes=true",
+    "--iree-opt-outer-dim-concat=true",
+    "--iree-rocm-waves-per-eu=2",
+    "--iree-llvmgpu-enable-prefetch",
+    "--iree-flow-enable-aggressive-fusion",
+    "--iree-global-opt-enable-fuse-horizontal-contractions=true",
+    "--iree-opt-aggressively-propagate-transposes=true",
+    "--iree-codegen-llvmgpu-use-vector-distribution=true",
+    "--iree-execution-model=async-external",
+    "--iree-preprocessing-pass-pipeline=builtin.module(iree-preprocessing-transpose-convolution-pipeline, util.func(iree-preprocessing-pad-to-intrinsics{pad-target-type=conv}))",
+]
+
+###############################################################################
+# CPU
+###############################################################################
+
+
+def test_compile_clip_cpu(sd3_clip_mlir):
+    VmfbManager.sd3_clip_cpu_vmfb = iree_compile(
+        sd3_clip_mlir, "cpu", CPU_COMPILE_FLAGS
+    )
+
+
+@pytest.mark.depends(on=["test_compile_clip_cpu"])
+def test_run_clip_cpu(SD3_CLIP_COMMON_RUN_FLAGS, sd3_clip_real_weights):
+    iree_run_module(
+        VmfbManager.sd3_clip_cpu_vmfb,
+        device="local-task",
+        function="encode_tokens",
+        args=[
+            f"--parameters=model={sd3_clip_real_weights.path}",
+            "--expected_f32_threshold=0.15f",
+        ]
+        + SD3_CLIP_COMMON_RUN_FLAGS,
+    )
+
+
+###############################################################################
+# ROCM
+###############################################################################
+
+
+@pytest.mark.xfail(
+    strict=True,
+    reason="Expected compilation to fail (remove xfail for test_compile_clip_rocm)",
+)
+def test_compile_clip_rocm(sd3_clip_mlir):
+    VmfbManager.sd3_clip_rocm_vmfb = iree_compile(
+        sd3_clip_mlir, f"rocm_{rocm_chip}", ROCM_COMPILE_FLAGS
+    )
+
+
+@pytest.mark.depends(on=["test_compile_clip_rocm"])
+def test_run_clip_rocm(SD3_CLIP_COMMON_RUN_FLAGS, sd3_clip_real_weights):
```

Comment:
Should add pytest marks for cpu, rocm, etc.

Reply:
Currently, we are using -k command line arg similar to SHARK-TestSuite/iree_tests

Reply:
Yeah, fine to do what is most similar to the current setup in this PR, but marks will be more flexible. We couldn't as easily use marks before since the test cases were generated.